pkg/api/validation/dynakube/oneagent_test.go

```diff
@@ -477,3 +477,49 @@ func createDynakubeWithHostGroup(args []string, hostGroup string) *dynakube.Dyna
 		},
 	}
 }
+
+func TestIsOneAgentVersionValid(t *testing.T) {
+	dk := dynakube.DynaKube{
+		ObjectMeta: defaultDynakubeObjectMeta,
+		Spec: dynakube.DynaKubeSpec{
+			APIURL: testApiUrl,
+			OneAgent: dynakube.OneAgentSpec{
+				ClassicFullStack: &dynakube.HostInjectSpec{},
+			},
+		},
+	}
+
+	t.Run("valid OneAgent custom versions are allowed", func(t *testing.T) {
+		versions := []string{"", "1.0.0.20240101-000000"}
```

Comment:
please add a test for a short version, e.g. `1.0.0`, as well!

Reply:
Short versions are invalid. They fail on OA DaemonSet level (with `version malformed`), that's why this validator rejects them too.

Reply:
https://github.com/Dynatrace/dynatrace-operator/pull/3966/files#diff-101115bf2e0d1f10e1c5efea14cf1d120ff0300df964b0883577c6fbbe84f25eR503